### PR TITLE
python3Packages.setuptools-rust: 0.11.6 -> 0.12.1

### PR DESCRIPTION
**Pull request by danieldk**
###### Motivation for this change

Changelog:

https://github.com/PyO3/setuptools-rust/releases/tag/v0.12.1
https://github.com/PyO3/setuptools-rust/releases/tag/v0.12.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 116954` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>home-assistant</li>
</ul>
</details>

File: pkgs/development/python-modules/setuptools-rust/default.nix [hunk 1]
```diff
@@ -11,12 +11,12 @@
 
 buildPythonPackage rec {
   pname = "setuptools-rust";
-  version = "0.11.6";
+  version = "0.12.1";
   disabled = isPy27;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/setuptools-rust/default.nix:15]: > Bump minimum Python version to Python 3.6.

Not sure if we should set it to disabled for python < 3.6.

**Review by SuperSandro2000 — APPROVED** [review]
LGTM

I just left some ideas to make improve python rust support.

Would be using this ``https://github.com/PyO3/setuptools-rust/issues/112`` feature make the hook easier?

Maybe cross compilation could benefit from ``https://github.com/PyO3/setuptools-rust/pull/96``?

**Comment by danieldk** [comment]
> Would be using this `https://github.com/PyO3/setuptools-rust/issues/112` feature make the hook easier?

Doesn't really affect `setuptools-rust` itself, which doesn't use any hook (outside the normal Python hooks). If projects upload sources to PyPI with vendored crates, then the corresponding derivations should not need `cargoSetupHook`. But that was already true if a project vendored their crates.

> Maybe cross compilation could benefit from `https://github.com/PyO3/setuptools-rust/pull/96`?

Could be. Unfortunately, the last time I tested, cross-compilation failed far before the build would actually a get to a `setuptools-rust` or `maturin` based package. But maybe `master` was just in a bad state at that point.

File: pkgs/development/python-modules/setuptools-rust/default.nix (hunk 1 above)
Comment (danieldk) [reply pkgs/development/python-modules/setuptools-rust/default.nix:15]: Although we don't have Python >3 && < 3.6 anymore, I don't think it hurts. Who knows, maybe someone has an old fork of nixpkgs and is overlaying newer packages ;). I'll update the PR.

Comment (danieldk) [reply pkgs/development/python-modules/setuptools-rust/default.nix]: Fixed. Also fixed the passthru test.

**Comment by SuperSandro2000** [comment]
@ofborg test python3Packages.setuptools-rust python38Packages.setuptools-rust python39Packages.setuptools-rust

**Comment by danieldk** [comment]
@ofborg build python37Packages.setuptools-rust python38Packages.setuptools-rust python39Packages.setuptools-rust

**Comment by danieldk** [comment]
@ofborg build python37Packages.setuptools-rust.passthru.tests python38Packages.setuptools-rust.passthru.tests python39Packages.setuptools-rust.passthru.tests

**Pull request merged**